pupil_remote/remote_annotations.py

```diff
@@ -45,20 +45,31 @@ def notify(notification):
         socket.send(payload)
         return socket.recv_string()
 
-    def send_trigger(label, timestamp, duration=0.):
-        return notify({'subject': 'annotation', 'label': label,
-                      'timestamp': timestamp, 'duration': duration,
-                      'source': 'example_script','record': True})
+    def send_trigger(label, timestamp, duration=0., **custom_keys):
+        minimal_trigger = {'subject': 'annotation', 'label': label,
+                           'timestamp': timestamp, 'duration': duration,
+                           'record': True}
+        minimal_trigger.update(custom_keys)
+        return notify(minimal_trigger)
 
     # Start the annotations plugin
     notify({'subject': 'start_plugin',
             'name': 'Annotation_Capture',
             'args': {}})
 
-    sleep(5.)  # sleep for a few seconds, can be less
+    socket.send_string('R')
+    socket.recv_string()
+
+    sleep(1.)  # sleep for a few seconds, can be less
 
     # Send a trigger with the current time
     # The annotation will be saved to the pupil_data notifications if a
     # recording is running. The Annotation_Player plugin will automatically
     # retrieve, display and export all recorded annotations.
-    send_trigger('custom trigger name', timestamp=time_fn())
```

Comment:
now that I see this, what about getting rid of the send trigger fn and just send a notification with the trigger dict?

Reply:
we can send a minimal trigger first, and then show how you can add custom keys.